go/mysql/datetime/parse.go

```diff
@@ -168,15 +177,31 @@ func clampTimeparts(tp *timeparts) bool {
 		tp.hour = 22
 		tp.min = 59
 		tp.sec = 59
-		return false
+		if state == TimeOK {
+			return TimePartial
+		}
 	}
-	return true
+	return state
 }
 
-func ParseTime(in string, prec int) (t Time, l int, ok bool) {
+type TimeState uint8
```

Comment:
minor nit: `datetime.TimeState` stutters. It should probably be `ParseState`

Reply:
Parsestate doesn’t communicate though this is only relevant for time parsing and not date or date time. So probably prefer this for now.